### PR TITLE
Add Caching Guide[WIP]

### PR DESCRIPTION
**Pull request by karansohi**
##### Checklist

- [ ] Tested in playground or other setup
- [ ] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

## Summary by CodeRabbit

- **Documentation**
  - Updated the sidebar positioning for the "API Quota Management" guide for better accessibility.
  - Added a new "Caching" guide with instructions and examples on how to use caching with the Aperture SDK.

- **New Features**
  - Introduced a new example module demonstrating caching capabilities with the Aperture SDK.

### CODE REVIEW

**Review by jaidesai-fn — APPROVED** [review]

File: docs/content/guides/caching.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import TabItem from "@theme/TabItem";
+import {BashTab, TabContent} from './blueprintsComponents.js';
+import CodeSnippet from '../codeSnippet.js'
+
+```
+
+## Overview
+
+Caching is a critical technology that significantly speeds up applications by
+storing frequently accessed data for quick retrieval. This approach reduces the
```
Comment (tanveergill) [inline docs/content/guides/caching.md:23]: Should mention that caching allows storing frequently accessed data in memory for quick retrieval

File: docs/content/guides/caching.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+leading to significant cost savings and performance improvements.
+
+<Zoom>
+
+```mermaid
+{@include: ./assets/caching/caching.mmd}
+```
+
+The diagram above shows how developers using the Aperture SDK can connect to
+Aperture Cloud, to set the cache or lookup a store response before processing an
```
Comment (tanveergill) [inline docs/content/guides/caching.md:43]: stored*

**Review by tanveergill — APPROVED** [review]

File: docs/content/guides/caching.md (hunk 1 above)
Comment (karansohi) [reply docs/content/guides/caching.md]: Done

File: docs/content/guides/caching.md (hunk 2 above)
Comment (karansohi) [reply docs/content/guides/caching.md]: Done

**Pull request merged**